smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -180,7 +177,7 @@ void execute() {
         // Generate models that are connected to the service being generated.
         LOGGER.fine("Walking shapes from " + service.getId() + " to find shapes to generate");
         // Walk the tree.
-        Collection<Shape> shapeSet = new Walker(nonTraits).walkShapes(service);
+        Collection<Shape> shapeSet = new Walker(model).walkShapes(service);
 
         Model prunedModel = Model.builder().addShapes(shapeSet).build();
```

Comment:
Is this necessary anymore? It was based upon the `nonTraits` model previously but that's gone, meaning places where this was used can just use `model` instead.

Reply:
Still needed. Otherwise, the TopologicalIndex will iterate and visit through more than `shapeSet`.

Reply:
Or I'll change it to use the `model` and check if shape is present in `shapeSet` before visiting.